### ISSUE TITLE
Is it possible to override the zod type with a speficic TypeScript type?

### ISSUE THREAD

**Issue by Greenheart** [efstajas/svelte-stored-writable#2]
Using the automatically inferred zod schema type is a good way to get started with this library. However, in some cases it's important to use specific runtime types to make the application easier to understand.

Is it possible to override the zod type with a speficic TypeScript type? For example passing in a generic type when creating the store, which would change the type definition into this:

```ts
storedWritable<T>(): Writable<T> & { clean: () => void } 
```

## Workaround

Meanwhile, I've used explicit type casting on the final end to work around this. This allows using zod to be used in the implementation, while keeping the same interface towards the rest of the application.

```ts
import type { Writable } from 'svelte/store'
import storedWritable from '@efstajas/svelte-stored-writable'
import { z } from 'zod'

export type Entry = {
    text: string
    time: Date
}

const schema = z.array(
    z.object({
        text: z.string(),
        time: z.date(),
    }),
)

export const entries = storedWritable(
    'entries',
    schema,
    [],
) as Writable<Entry[]> & { clear: () => void }
```

No worries if this is out of scope for this project. Just raising it in case someone else wonders the same thing.

**Comment by efstajas** [comment]
Good idea! I'd set the type param to match the zod-inferred type by default, and constrain it so that its fields match the zod schema, in order to avoid any mismatches. Will add this soon.

**Comment by efstajas** [comment]
@Greenheart Just published `v0.1.3`, which includes the ability to pass a type parameter to `storedWritable`: https://github.com/efstajas/svelte-stored-writable/tree/v0.1.3#setting-a-custom-writable-type

Lmk if this suits your needs!
